File: src/main/java/com/ecommerce/handcraft/requests/CreateProductRequest.java
package com.ecommerce.handcraft.requests;

import javax.validation.constraints.NotNull;
import java.util.List;

public class CreateProductRequest {
    private String productName;
    private String productDescription;
    private Integer price;
    private Integer discount;
    private Long categoryId;
    private List<Long> colorIds;
    private List<String> imageUrls;

    public List<String> getImageUrls() {
        return imageUrls;
    }

    public String getProductName() {
        return productName;
    }

    public String getProductDescription() {
        return productDescription;
    }

    public Integer getPrice() {
        return price;
    }

    public Integer getDiscount() {
        return discount;
    }

    public Long getCategoryId() {
        return categoryId;
    }

    public List<Long> getColorIds() {
        return colorIds;
    }
}

File: src/main/handcraft-react/src/Models/Requests/CreateProductRequest.ts
class CreateProductRequest{
    productName:string;
    productDescription:string;
    price:number;
    discount:number;
    categoryId:number;
    colorIds:number[];
    imageUrls:string[];
    constructor(   productName:string,
        productDescription:string,
        price:number,
        discount:number,
        categoryId:number,
        colorIds:number[],
        imageUrls:string[]){
            this.productName=productName;
            this.productDescription=productDescription;
            this.price=price;
            this.discount = discount;
            this.categoryId = categoryId;
            this.colorIds = colorIds;
            this.imageUrls=imageUrls
        }
}
export default CreateProductRequest